src/Quadrature/quadrature.jl

```diff
@@ -42,18 +42,19 @@ julia> getpoints(qr)
  [0.33333333333333, 0.33333333333333]
 ```
 """
-struct QuadratureRule{shape,T,dim}
-    weights::Vector{T}
-    points::Vector{Vec{dim,T}}
-    function QuadratureRule{shape, T}(weights::Vector{T}, points::Vector{Vec{dim, T}}) where {dim, shape <: AbstractRefShape{dim}, T}
+# struct QuadratureRule{shape, T, rdim, WeightStorageType <: AbstractVector{T}, PointStorageType <: AbstractVector{<:Vec{rdim,T}}}
+struct QuadratureRule{shape, T, rdim, WeightStorageType <: AbstractVector{T}, PointStorageType <: AbstractVector} # The one above crashes the deprecation in deprecations.jl for some reason...
+    weights::WeightStorageType
```

Comment:
Why do they need to be able to have different storage type and different array type? Couldn't the change here only allow for swapping out the `Vector` part while keeping the rest?

Reply:
Fredrik just asked me the same :D I do not know how, because the vectors store different things. I may be able to store some compressed variant of the Vec (as a AbstractVector{T} with num weights * dim entries), but not sure if it is worth the effort.

Reply:
An alternative is to have an `AbstractQuadratureRule` that needs to support some interface and then if you need really exotic ones you can always BYOQR (Bring Your Own Quadrature Rule). And then the FEValues would just take a `QR <: AbstractQuadratureRule`.